Add more gradients examples

jmetaxas opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

- Added some examples of gradients using the **background-blend-mode** CSS property.
- Added an example of a repeating conic gradient.
- Added an example of multiple repeating conic gradients.
- Added a link to a css gradient generator that supports the use of layers and blend modes to create [advanced effects](https://colorbeta.com/gradients).

<!-- ✍️ Summarize your changes in one or two sentences -->

### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->

The page lacked examples using the **repeating-conic-gradient()** CSS function, as well as the ability to create more advanced effects with the **background-blend-mode** property.

### Additional details

<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->

### Related issues and pull requests

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

jmetaxas committed:
Adding repeating conic examples

jmetaxas committed:
Merge branch 'main' into update-gradients

jmetaxas committed:
Adding examples with background-blend-mode

jmetaxas committed:
Merge branch 'main' into update-gradients

estelle commented on files/en-us/web/css/css_images/using_css_gradients/index.md:
```diff
@@ -923,6 +968,67 @@ div {
+
+### Multiple repeating conic gradients
+
+Just like linear and radial repeating gradients, you can stack multiple conic gradients on top of each other. You will need to ensure either that the colors of the gradients on the top of the stack are partially transparent or use the {{cssxref("background-blend-mode", "background-blend-mode")}} CSS property
```
```suggestion
Just like linear and radial repeating gradients, you can stack multiple conic gradients on top of each other, creating interesting effects by using different `at <position>` values so the conic gradients don't overlap at their centers and different `from <angle>` values so the repeating effects don't line up. This example overlaps three semi-transparent repeating radial gradients that each repeat their color schemes four times. You will need to ensure either that the colors of the gradients on the top of the stack are partially transparent or use the {{cssxref("background-blend-mode")}} CSS property.
```
* the second parameter of the cssxref macro is only needed if the text is different.
* mentioning this cool effect is created by putting the center of each conic gradient in a different location.

estelle commented on files/en-us/web/css/css_images/using_css_gradients/index.md:
```diff
@@ -923,6 +968,67 @@ div {
 
+### Repeating conic gradients
+
+This example uses {{cssxref("gradient/repeating-conic-gradient", "repeating-conic-gradient()")}} to create a gradient that rotates repeatedly around a center point. The colors get cycled over and over as the gradient repeats.
```
```suggestion
This example uses {{cssxref("gradient/repeating-conic-gradient", "repeating-conic-gradient()")}} to create a gradient that rotates repeatedly around a center point. In this case, the declared color stops are repeated four times.
```

estelle commented on files/en-us/web/css/css_images/using_css_gradients/index.md:
```diff
@@ -362,6 +362,51 @@ div {
 
 {{ EmbedLiveSample('Stacked_gradients', 200, 200) }}
 
+### Blending gradients
```
This example uses gradients that aren't yet introduced. We've only seen linear gradients thus far. To create a useful example, limit the example to features that are already discussed plus the feature that is being added by the current example.

estelle commented on files/en-us/web/css/css_images/using_css_gradients/index.md:
````diff
@@ -362,6 +362,51 @@ div {
+In addition to transparency, stacked gradients can also be blended with the {{cssxref("background-blend-mode", "background-blend-mode")}} CSS property.
+
+```html hidden
+<div class="lighten-gradients"></div>
````
include an "original" with no background-blend mode or any effects.

estelle commented on files/en-us/web/css/css_images/using_css_gradients/index.md:
```diff
@@ -362,6 +362,51 @@ div {
 
+### Blending gradients
+
+In addition to transparency, stacked gradients can also be blended with the {{cssxref("background-blend-mode", "background-blend-mode")}} CSS property.
```
```suggestion
In addition to transparency, stacking multiple semi-transparent gradients and stacking gradients over raster background images, gradients can be used with other CSS effects. In this example, we use the {{cssxref("background-blend-mode")}} CSS property to create blended effects.
```
background-blend-mode isn't directly related to gradients. Trying to make this more relevant to the page (though it still has the "we having seen these types of gradients yet" issue.

estelle reviewed (commented):
Thanks for the PR. I provided some feedback to help make your suggested changes flow better within the content of the current guide.

jmetaxas committed:
Merge branch 'main' into update-gradients

jmetaxas committed:
Implemented suggested changes for Conic and Repeating Conic sections

jmetaxas committed:
Simplified blending gradients examples

jmetaxas commented:
Hi Estelle, 

Thank you very much for you feedback. I've implemented the changes you suggested. 

You are absolutely right about the "blending gradients" examples, they were a little too complex, especially since the article had only just started discussing linear gradients... I've simplified them, by keeping just two stacked linear gradients, each with two different colors.

jmetaxas committed:
Merge branch 'main' into update-gradients

estelle commented on files/en-us/web/css/css_images/using_css_gradients/index.md:
````diff
@@ -362,6 +362,50 @@ div {
+In addition to transparency, stacking multiple semi-transparent gradients and stacking gradients over raster background images, gradients can be used with other CSS effects. In this example, we use the {{cssxref("background-blend-mode")}} CSS property to create blended effects.
+
+```html hidden
+<div class="normal-gradients"></div>
+<div class="screen-gradients"></div>
+<div class="overlay-gradients"></div>
+<div class="difference-gradients"></div>
````
```suggestion
<div class="original"></div>
<div class="screen"></div>
<div class="overlay"></div>
<div class="difference"></div>
```

estelle commented on files/en-us/web/css/css_images/using_css_gradients/index.md:
````diff
@@ -362,6 +362,50 @@ div {
+```
+
+```css
+.normal-gradients {
+  background: linear-gradient(to top, red, blue), linear-gradient(to right, #5500ff, #00ff55);
+}
+
+.screen-gradients {
+  background: linear-gradient(to top, red, blue), linear-gradient(to right, #5500ff, #00ff55);
+  background-blend-mode: screen;
+}
+
+.overlay-gradients {
+  background: linear-gradient(to top, red, blue), linear-gradient(to right, #5500ff, #00ff55);
+  background-blend-mode: overlay;
+}
+
+.difference-gradients {
+  background: linear-gradient(to top, red, blue), linear-gradient(to right, #5500ff, #00ff55);
+  background-blend-mode: difference;
+}
````
```suggestion
div {
  background: linear-gradient(to top, red, blue), linear-gradient(to right, #5500ff, #00ff55);
}

.screen {
  background-blend-mode: screen;
}

.overlay{
  background-blend-mode: overlay;
}

.difference {
  background-blend-mode: difference;
}
```

estelle commented on files/en-us/web/css/css_images/using_css_gradients/index.md:
```diff
@@ -362,6 +362,50 @@ div {
 
+### Blending gradients
+
+In addition to transparency, stacking multiple semi-transparent gradients and stacking gradients over raster background images, gradients can be used with other CSS effects. In this example, we use the {{cssxref("background-blend-mode")}} CSS property to create blended effects.
```
```suggestion
In addition to transparency, stacking multiple semi-transparent gradients and stacking gradients over raster background images, gradients can be used with other CSS effects. In this example, the four {{htmlelement("div")}} elements have the same gradient as a background image and we apply different {{cssxref("background-blend-mode")}} CSS property values to the last three to create blended effects.
```

estelle commented on files/en-us/web/css/css_images/using_css_gradients/index.md:
```diff
@@ -923,6 +967,67 @@ div {
+
+### Multiple repeating conic gradients
+
+Just like linear and radial repeating gradients, you can stack multiple conic gradients on top of each other, creating interesting effects by using different `at <position>` values so the conic gradients don't overlap at their centers and different `from <angle>` values so the repeating effects don't line up. This example overlaps three semi-transparent repeating radial gradients that each repeat ...
```
```suggestion
Just like linear and radial repeating gradients, you can stack multiple conic gradients on top of each other, creating interesting effects by using different `at <position>` values so the conic gradients don't overlap at their centers and different `from <angle>` values so the repeating effects don't line up. This example overlaps three semi-transparent repeating radial gradients that each repeat their color schemes four times. To make overlapping gradients visible, you need to ensure either that the colors of the gradients on the top of the stack are partially transparent or use the {{cssxref("background-blend-mode")}} CSS property.
```

estelle reviewed (commented):
I made a suggestion to change the way the code is presented. I changed the class names to the value of the blend mode property and suggesting making the HTML visible so readers know which is which. I put the gradient on the div, no class name, so that it is apparent that they are all the same gradient. 

We're close!

jmetaxas committed:
Merge branch 'main' into update-gradients

jmetaxas committed:
simplified background-blend-mode examples

jmetaxas committed:
rewording of the 'blending gradients' and 'multiple repeating conic gradients' paragraphs

jmetaxas commented:
Hi @estelle,

I wasn't sure if it was advisable to simplify the syntax of the examples, but it is indeed much clearer and more readable this way.

estelle reviewed (approved):
Thank you. 

Congratulations on your first merged MDN content PR. 
Welcome to the team!

-Estelle


Tip for a faster review: When you are done making changes after a PR review and want a re-review, click on the icon next the reviewer's name. There will be a double grey arrow, as in the attached screen shot, next to a speech bubble or a red icon that looks like a sheet of paper with a + and - on it. Once you click on it, it becomes a brown dot. Doing this sends a notification to the reviewer that the PR is ready for a re-review.

estelle commented:
> the attached screen shot, 
<img width="347" alt="Screenshot 2024-04-18 at 2 33 14 PM" src="https://github.com/mdn/content/assets/69888/ce4a842e-14a0-48a2-b7a4-472ffa73dbc8">
<img width="350" alt="Screenshot 2024-04-18 at 2 31 24 PM" src="https://github.com/mdn/content/assets/69888/93d61ddc-82ba-4502-8d1a-f762008b57dd">
<img width="354" alt="Screenshot 2024-04-18 at 2 31 15 PM" src="https://github.com/mdn/content/assets/69888/e64f8fe9-986c-4bd4-ad0e-78424becd4d9">

estelle commented on files/en-us/web/css/css_images/using_css_gradients/index.md:
```diff
@@ -362,6 +362,47 @@ div {
 
+### Blending gradients
+
+In addition to transparency, stacking multiple semi-transparent gradients and stacking gradients over raster background images, gradients can be used with other CSS effects. In this example, the four {{htmlelement("div")}} elements have the same gradient as a background image and we apply different {{cssxref("background-blend-mode")}} CSS property values to the last three to create blended effect ...
```
```suggestion
In addition to transparency, stacking multiple semi-transparent gradients and stacking gradients over raster background images, gradients can be used with other CSS effects. In this example, the four {{htmlelement("div")}} elements have the same two fully-opaque gradients as background images. We apply different {{cssxref("background-blend-mode")}} CSS property values to the last three that blend the two background images creating different effects.
```

estelle committed:
Update files/en-us/web/css/css_images/using_css_gradients/index.md

jmetaxas commented:
Thank you for your kind and helpful advice, @estelle.
